packages/fxa-payments-server/src/routes/Product/index.tsx

```diff
@@ -237,15 +268,35 @@ export const Product = ({
       productId
     );
 
-    if (iapSubscription) {
+    // Note regarding IAP roadblock:
+    //
+    // Currently, customers cannot upgrade IAP subscriptions directly.
+    // Products that are eligible for upgrade can only be done through support.
+    //
+    // iapSubscription checks if user is subscribed to product (not product set)
+    // planUpgradeEligibility returns 'blocked_iap' for all products on same product set.
+    // Therefore, a combination of both is required to show IAP roadblock (planUpgradeEligibility)
+    // and appropriate error messaging (iapSubscription).
+    //
+    // if desired product is already subscribed to, show iap already subscribed error
+    // else, product is not subscribed to, but on same product set/might be eligible for upgrade
+    // show iap upgrade contact support error messaging
+    const iapErrorMessageCode = () => {
```

Comment:
nit. Does this need to be a function call? Could probably just in line this where it's used.